### PR TITLE
Feat: Pages with `RenderSection` + `Breadcrumb` section CMS

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

You may notice that in some pages (e.g. PDP) the breadcrumb is not shown. That's because CMS team is investigating some behaviors and maybe the Breadcrumb section [is not present in CMS](https://storeframework.myvtex.com/admin/new-cms/faststore/pdp/edit/717bc97f-8f19-4744-a52b-7ccac43cc898/).

- [x] Adjusts the `Breadcrumb` section to be rendered using CMS data.
- [x] Moves `Breadcrumb` from `ProductDetails` to `PDP` page.
- [x] Adapts `Breadcrumb` in `PDP`, `PLP` and `Search` pages.
- [x] Prepare pages to be rendered using `RenderSection` with CMS data (CMS).
- [x] Adds Search content-type.
- [x] Promise all sequential requests from server side (static) pages (performance: build time).

## How it works?

It gets data from CMS and pass to `RenderSections` components, which in turn pass to `Breadcrumb` section. As we have this component in some pages, we also pass a `context` prop depending on the page:
- PLP - context = collection
- PDP - context = product
- Search - context = undefined.

## How to test it?

run `yarn dev` into the core package and check the breadcrumb in these pages.
you can also see the starter preview.

### Starters Deploy Preview

PR https://github.com/vtex-sites/starter.store/pull/54
Preview https://preview-54--starter.preview.vtex.app/

### CODE REVIEW

**Review by ArthurTriis1 — APPROVED** [review]
LGTM! I just realized the PDP doesn't have Breadcrumb, is it right?

**Comment by eduardoformiga** [comment]
> 

Nop, The breadcrumb should be shown in PDP. Someone removed it from CMS. I'll place it there again ;)

File: packages/core/src/pages/index.tsx [hunk 1]
```diff
@@ -85,14 +85,15 @@ export const getStaticProps: GetStaticProps<
...
-  const globalSections = await getGlobalSectionsData(previewData)
+  const [page, globalSections] = await Promise.all([
+    getPage<PageContentType>({
+      ...(previewData?.contentType === 'page'
+        ? previewData
+        : { filters: { 'settings.seo.slug': '/' } }),
+      contentType: 'page',
+    }),
+    getGlobalSectionsData(previewData),
+  ])
```
Comment (lucasfp13) [inline packages/core/src/pages/index.tsx:96]: Nice 👍🏼

File: packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -4,28 +4,26 @@ import { memo } from 'react'
...
+}: BreadcrumbProps) => (
   <UIBreadcrumb
-    breadcrumbList={breadcrumbList}
     homeLink={
-      <Link aria-label="Go to homepage" href="/" prefetch={false}>
-        <Icon name="House" width={18} height={18} weight="bold" />
+      <Link
+        data-fs-breadcrumb-link
+        data-fs-breadcrumb-link-home
+        aria-label={`${alt}`}
```
Comment (lucasfp13) [inline packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx:22]: The `alt` incoming from CMS is a string, maybe we don't need this concatenation here, WDYT?

File: packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx [hunk 3]
```diff
@@ -4,28 +4,26 @@ import { memo } from 'react'
...
+        href="/"
+        prefetch={false}
+      >
+        <Icon name={`${icon}`} width={18} height={18} weight="bold" />
```
Comment (lucasfp13) [inline packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx:26]: The same regarding the `icon` here being a string

**Review by lucasfp13 — APPROVED** [review]
LGTM! 🚀 
Left just some comments.

File: packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (eduardoformiga) [reply packages/core/src/components/ui/Breadcrumb/Breadcrumb.tsx]: nice catch! done [here](https://github.com/vtex/faststore/pull/1747/commits/8178118bfa69ddcb39e0eefc106271853b662b87).

**Pull request merged**